tailscale/client.go

```diff
@@ -937,6 +937,160 @@ func (c *Client) SetDeviceIPv4Address(ctx context.Context, deviceID string, ipv4
 	return c.performRequest(req, nil)
 }
 
+const (
+	EmptyProviderType      ProviderType = ""
+	SlackProviderType      ProviderType = "slack"
+	MattermostProviderType ProviderType = "mattermost"
+	GoogleChatProviderType ProviderType = "googlechat"
+	DiscordProviderType    ProviderType = "discord"
+)
+
+const (
+	NodeCreated                    SubscriptionType = "nodeCreated"
+	NodeNeedsApproval              SubscriptionType = "nodeNeedsApproval"
+	NodeApproved                   SubscriptionType = "nodeApproved"
+	NodeKeyExpiringInOneDay        SubscriptionType = "nodeKeyExpiringInOneDay"
+	NodeKeyExpired                 SubscriptionType = "nodeKeyExpired"
+	NodeDeleted                    SubscriptionType = "nodeDeleted"
+	PolicyUpdate                   SubscriptionType = "policyUpdate"
+	UserCreated                    SubscriptionType = "userCreated"
+	UserNeedsApproval              SubscriptionType = "userNeedsApproval"
+	UserSuspended                  SubscriptionType = "userSuspended"
+	UserRestored                   SubscriptionType = "userRestored"
+	UserDeleted                    SubscriptionType = "userDeleted"
+	UserApproved                   SubscriptionType = "userApproved"
+	UserRoleUpdated                SubscriptionType = "userRoleUpdated"
+	SubnetIPForwardingNotEnabled   SubscriptionType = "subnetIPForwardingNotEnabled"
+	ExitNodeIPForwardingNotEnabled SubscriptionType = "exitNodeIPForwardingNotEnabled"
+)
+
+type (
+	// ProviderType defines the provider type for a Webhook destination.
+	ProviderType string
+
+	// SubscriptionType defines events in tailscale to subscribe a Webhook to.
+	SubscriptionType string
+
+	// Webhook type defines a webhook endpoint within a tailnet.
+	Webhook struct {
+		EndpointID       string             `json:"endpointId"`
+		EndpointURL      string             `json:"endpointUrl"`
+		ProviderType     ProviderType       `json:"providerType"`
+		CreatorLoginName string             `json:"creatorLoginName"`
+		Created          time.Time          `json:"created"`
+		LastModified     time.Time          `json:"lastModified"`
+		Subscriptions    []SubscriptionType `json:"subscriptions"`
+		Secret           string             `json:"secret,omitempty"`
+	}
+
+	// CreateWebhookRequest type describes the configuration for creating a Webhook.
+	CreateWebhookRequest struct {
+		EndpointURL   string             `json:"endpointUrl"`
+		ProviderType  ProviderType       `json:"providerType"`
+		Subscriptions []SubscriptionType `json:"subscriptions"`
+	}
+)
+
+// CreateWebhook creates a new webhook with the specifications provided in the CreateWebhookRequest.
+// Returns a Webhook if successful.
+func (c *Client) CreateWebhook(ctx context.Context, request CreateWebhookRequest) (Webhook, error) {
+	const uriFmt = "/api/v2/tailnet/%s/webhooks"
+
+	req, err := c.buildRequest(ctx, http.MethodPost, fmt.Sprintf(uriFmt, c.tailnet), requestBody(request))
+	if err != nil {
+		return Webhook{}, err
```

Comment:
In at least one other place, I see us using the pointer return style to allow just returning `nil`.

```
func (c *Client) DeviceSubnetRoutes(ctx context.Context, deviceID string) (*DeviceRoutes, error) {
```

We also have other places where we return arrays, which also allows returning `nil` on errors.

I personally prefer that style.

Reply:
Yeah that is fair! Will move over to that style.